src/internal/connector/onedrive/permission.go

```diff
@@ -194,7 +198,17 @@ func restorePermissions(
 	permAdded, permRemoved := diffPermissions(currentPermissions, meta.Permissions, permissionIDMappings)
 
 	for _, p := range permRemoved {
-		err := service.Client().
+		// deletes require unique http clients
+		// https://github.com/alcionai/corso/issues/2707
+		// this is bad citizenship, and could end up consuming a lot of
+		// system resources if servicers leak client connections (sockets, etc).
+		a, err := graph.CreateAdapter(m365.AzureTenantID, m365.AzureClientID, m365.AzureClientSecret)
```

Comment:
@meain I'm a bit concerned about this point, since it feels like we might accumulate a lot of calls during a restore runtime.  Do you know any expectations about how frequently this could get invoked?

Reply:
I feel like this should be pretty limited in most normal cases.